InAppUtils/InAppUtils.m

```diff
@@ -19,13 +20,33 @@ - (instancetype)init
     return self;
 }
 
+-(void)startObserving {
+    hasPurchaseCompletedListeners = YES;
+}
+
+-(void)stopObserving {
+    hasPurchaseCompletedListeners = NO;
+}
+
 - (dispatch_queue_t)methodQueue
 {
     return dispatch_get_main_queue();
 }
 
 RCT_EXPORT_MODULE()
 
+- (NSArray<NSString *> *)supportedEvents
+{
+    return @[@"PurchaseCompleted"];
+}
+
+// Transactions initiated from App Store
+- (BOOL)paymentQueue:(SKPaymentQueue *)queue
+shouldAddStorePayment:(SKPayment *)payment
+          forProduct:(SKProduct *)product {
+    return true;
```

Comment:
can we use `hasPurchaseCompletedListeners`?

Reply:
Done!